### DISCUSSION TITLE
Create Site Collection - Unattended

### DISCUSSION THREAD

**Discussion by anthonywhite** [pnp/pnpframework#289] [Q&A] [answered]
I am looking to automate Site Collection creation in a **C#** Azure Function, authenticating app-only with the app registered through the `Register-PnPAzureADApp` PowerShell command. I based this on the [sample PnP Azure Function](https://pnp.github.io/pnpcore/demos/Demo.AzureFunction/README.html) and that all works fine when retrieving data from an existing site.

However I cannot see any method in `PnP.Framework` or `PnP.Core` that does Site creation using a `PnPContext`. Have I missed it somewhere? The `Framework.Sites.SiteCollection.CreateAsync()` requires a `Microsoft.SharePoint.Client.ClientContext` (as does the `Microsoft.Online.SharePoint.TenantAdministration.Tenant.CreateSiteCollection` method).

Is there no equivalent of the `New-PnPTenantSite` PowerShell command in the Framework? Can anyone suggest the best way of doing this in my function?

**Comment by jansenbe** [comment, marked as answer]
@anthonywhite : Site collection creation is done using REST API's for the most part, creating a group connected site using App-Only uses a Graph API call. Code lives here: https://github.com/pnp/pnpframework/blob/dev/src/lib/PnP.Framework/Sites/SiteCollection.cs

Currently we do not yet have support for creating site collections in PnP Core SDK, but as mentioned by you when using a CSOM client context this works...the CSOM client context is just used to obtain an access token for the REST/Graph calls. We do plan to add support for site collection creation in PnP Core SDK going forward, just did not get to it yet

**Reply by anthonywhite** [reply]
OK thanks Bert - I need non-group sites. Is there some utility that creates a ClientContext from a PnPContext? Or will I need to start over?

**Reply by jansenbe** [reply]
See answer to #294 , you can use that approach to issue the rest call that creates the site collection

**Reply by anthonywhite** [reply]
👍🏻
